File: PA100.py
#Time Complexity - O(n)
#Space Complexity - O(n)
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def kthSmallest(self, root, k):
        """
        :type root: TreeNode
        :type k: int
        :rtype: int
        """
        if not root:
            return 0
        self.ans = []
        def ino(root):
            if root:
                ino(root.left)
                self.ans.append(root.val)
                ino(root.right)
        ino(root)
        return self.ans[k-1]
            

File: PA100.java
//Leetcode :  230. Kth Smallest Element in a BST
//Time Complexity: O(n), n is no of nodes in the tree
//Space Complexity: O(n) 

class Solution {
    int result;
    int n;
    public int kthSmallest(TreeNode root, int k) {
        result=-1;
        n=k;
        helper(root);
        return result;
    }
    public void helper(TreeNode root){
        if(root==null) return ;      
        helper(root.left);
        n--;
       if(n==0){
            result=root.val;
            return;
        }
        helper(root.right);
    }
}